### ISSUE TITLE
reserve-with-timeout seem to be buggy

### ISSUE THREAD

**Issue by vimukthi-git** [vimukthi-git/beanstalkg#73]
I observed that reserve with time out produces inconsistent behaviour when clients reconnect. Needs further investigation

**Comment by stephenwithav** [comment]
Can you reproduce the bug?

**Comment by vimukthi-git** [comment]
yeah just run the server locally and run reserve with timeout from multiple clients. I can give you code to reproduce soon.
